### PR TITLE
Added "Spring Cloud Contract, meet Pact" post

### PR DESCRIPTION
**Pull request by TYsewyn**

### CODE REVIEW

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+tags: [Spring, Spring Cloud, Spring Cloud Contract, SCC, Pact, CDCT]
+category: Spring
+comments: true
+---
+
+# CDCT, or Consumer-Driven Contract Testing
+
+Consumer-Driven contract tests are actually integration tests that are targetting your API, whether it's REST-based or messaging-based.
+Key in this setup is that the tests are driven by contracts that are defined by the consumer of your API instead of the developer that wrote the implementation of a certain functionality.
+Using this kind of tests we can verify whether or not we're going to break some functionality from our consumer in case we apply some changes to our API.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: these kinds of tests

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+With Spring Cloud Contract we could already define groovy contracts which were packaged as a jar and uploaded to your artifact repository like Nexus or Artifactory.
+Great, but that meant that we could only use these contracts between JVM languages.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: The earlier sentence isn't the cause of the JVM limitation.
No support for SCS libraries in other languages (as opposed to Pact libraries), is.
Also, the word `between` isn't correct here. It should be `using` I suppose.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+What if the consumer of our API is a NodeJS microservice, or an Angular application?
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Perhaps add a third example which isn't using Javascript. Python, Ruby or whatever.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+# Spring Cloud Contract, meet Pact
+
+The first release candidate of Spring Cloud Finchley, which was released yesterday, ships also the first release candidate of Spring Cloud Contract.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: also ships

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Spring Cloud Contract has added the support to connect to a <a href="https://github.com/pact-foundation/pact_broker" target="_blank">Pact broker</a> to retrieve or store Pact contracts.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Perhaps add some information about the Pact Broker and why it's so awesome.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+The Pact contracts will be converted from and to Spring Cloud Contract contracts for you, so these in turn can be used to generate stubs and tests.
+
+Simply add the `spring-cloud-contract-pact` dependency, which will add the Pact Converter and Pact Stub Downloader, together with some configuration and you're good to go!
+
+Currently Pact contracts up until v4 are supported, which means that both request-response and messaging contracts can be used.
+Note that not all functionality is supported though.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Perhaps make a reference here to github issues or discussions outlining what is currently missing.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+category: Spring
+comments: true
+---
+
+# CDCT, or Consumer-Driven Contract Testing
+
+Consumer-driven contract tests are actually integration tests that are targetting your API, whether it's REST-based or messaging-based.
+Imagine you're working on an application that exposes its data using a REST API.
+Another team is using your exposed data for some functionality that they are providing.
+In order to guarantee that the functionality of the other team their application doesn't break if we make changes to our API, we create a contract between the 2 teams.
```
Comment (yannickdeturck) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: two teams

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+Key in this setup is that the contracts that are defined by the consumer of your API instead of the developer that wrote the implementation of a certain functionality.
```
Comment (yannickdeturck) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: ... is that the contracts are defined by the consumer ... (drop the "that")

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+With this aproach we can generate tests using the consumer-driven contracts (CDCs) and verify whether or not we're going to break some functionality from our consumer's application in case we apply some changes to our API.
+You can find a couple of interesting and useful links on the bottom of this post which go further into detail.
+
+With Spring Cloud Contract we could already define groovy contracts which were packaged as a jar and uploaded to your artifact repository like Nexus or Artifactory.
```
Comment (yannickdeturck) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Groovy

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+      <figcaption class="align-center">A graph showing the dependencies between all known consumers and producers</figcaption>
+    </a>
+  </div>
+</div>
+<br />
+
+
+Simply add the `spring-cloud-contract-pact` dependency, which will add the Pact Converter and Pact Stub Downloader, together with some configuration and you're good to go!
+
+The Pact contracts will be retrieved from the pact broker and converted to Spring Cloud Contract contracts for you, so these in turn can be used to generate stubs and tests.
```
Comment (yannickdeturck) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: pact -> Pact

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+The upside to it is that these contracts are in fact language agnostic as opposed to Spring Cloud Contract, the downside to it is that it lacks some functionality regarding messaging compared to Spring Cloud Contract.
+
+Currently Pact contracts up until v4 are supported, which means that both request-response and messaging contracts can be used.
+Note that not all functionality is supported though, as is described in <a href="https://cloud.spring.io/spring-cloud-contract/single/spring-cloud-contract.html#pact-converter" target="_blank">this section</a> of the reference documentation.
+
+A typical workflow to define such contracts could look like this:
+- The consumer of the data writes a test for the feature by doing TDD.
+In this test the contract is defined which will be used as a stub.
+- Next up the missing implementation is written.
+- When we push our changes to git our CI pipeline is triggered, in which we'll upload all defined contracts to a central artifact repository, eg. a Pact broker, after all our tests have been run.
```
Comment (yannickdeturck) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: git -> Git

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+---
+
+# CDCT, or Consumer-Driven Contract Testing
+
+Consumer-driven contract tests are actually integration tests that are targetting your API, whether it's REST-based or messaging-based.
+Imagine you're working on an application that exposes its data using a REST API.
+Another team is using your exposed data for some functionality that they are providing.
+In order to guarantee that the functionality of the other team their application doesn't break if we make changes to our API, we create a contract between the two teams.
+Key in this setup is that the contracts are defined by the consumer of your API instead of the developer that wrote the implementation of a certain functionality.
+With this aproach we can generate tests using the consumer-driven contracts (CDCs) and verify whether or not we're going to break some functionality from our consumer's application in case we apply some changes to our API.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Kind of weird sentence. Maybe remove "the"

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+You can find a couple of interesting and useful links on the bottom of this post which go further into detail.
+
+With Spring Cloud Contract we could already define Groovy contracts which were packaged as a jar and uploaded to your artifact repository like Nexus or Artifactory.
+Great, but that meant that we could only use these contracts between two parties that both are using a JVM language.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Still no cause in the previous sentence. Only jvm library support is the cause.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+<center><blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Our first release candidate of the Spring Cloud Finchley release train has been released.  Checkout the blog post for more information and as always feedback is welcome! <a href="https://t.co/8TK0tudkzr">https://t.co/8TK0tudkzr</a></p>&mdash; Spring Cloud (@springcloud) <a href="https://twitter.com/springcloud/status/989122422635925504?ref_src=twsrc%5Etfw">April 25, 2018</a></blockquote></center>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+<br />
+
+Spring Cloud Contract has updated its support for <a href="http://pact.io/" target="_blank">Pact</a> and added the support to connect to a <a href="https://github.com/pact-foundation/pact_broker" target="_blank">Pact broker</a>.
+The Pact broker acts as a repository for sharing the Pact contracts and verification results.
+The most awesome feature in my opinion is the visualisation of the contracts between all known parties.
+In the UI you can see the last time there was a new version of a contract published, when it was verified and what status it had, and last but not least the dependency graph.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Dependency graph was mentioned in the previous sentence.

File: _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+Simply add the `spring-cloud-contract-pact` dependency, which will add the Pact Converter and Pact Stub Downloader, together with some configuration and you're good to go!
+
+The Pact contracts will be retrieved from the Pact broker and converted to Spring Cloud Contract contracts for you, so these in turn can be used to generate stubs and tests.
+It's also possible to write Spring Cloud Contract contracts, convert them to Pact contracts and upload them to the Pact broker.
+
+The upside to it is that these contracts are in fact language agnostic as opposed to Spring Cloud Contract, the downside to it is that it lacks some functionality regarding messaging compared to Spring Cloud Contract.
+
+Currently Pact contracts up until v4 are supported, which means that both request-response and messaging contracts can be used.
+Note that not all functionality is supported though, as is described in <a href="https://cloud.spring.io/spring-cloud-contract/single/spring-cloud-contract.html#pact-converter" target="_blank">this section</a> of the reference documentation.
```
Comment (andreasevers) [inline _posts/2018-04-26-Spring-Cloud-Contract-meet-Pact.md]: Previous paragraph also mentioned lacking of functionality.

**Pull request merged**